src/songMetadata/musicBrainzModule/musicBrainzClient.ts

```diff
@@ -0,0 +1,117 @@
+    trackNumberFreeText = 'number', // free text track number
+    position = 'position', // the medium that the recording should be found on, first medium is position 1
+    primaryType = 'primarytype', // primary type of the release group (album, single, ep, other)
+    qdur = 'qdur', // quantized duration (duration / 2000)
+    recordingId = 'rid', // recording id
+    recordingName = 'recording', // name of recording or a track associated with the recording
+    recordingAccent = 'recordingaccent', // name of the recording with any accent characters retained
+    releaseId = 'reid', // release id
+    releaseName = 'release', // release name
+    releaseGroupId = 'rgid', // release group id
+    secondaryType = 'secondarytype', // secondary type of the release group (audiobook, compilation, interview, live, remix soundtrack, spokenword)
+    status = 'status', // Release status (official, promotion, Bootleg, Pseudo-Release)
+    trackId = 'tid', // track id
+    trackNumberOnMedium = 'tnum', // track number on medium
+    tracksCount = 'tracks', // number of tracks in the medium on release
+    tracksCountOnRelease = 'tracksrelease', // number of tracks on release as a whole
+    tag = 'tag', // folksonomy tag
+    type = 'type', // type of the release group, old type mapping for when we did not have separate primary and secondary types or use standalone for standalone recordings
+    video = 'video', // true to only show video tracks
+}
+
+export type SearchReleasesCriteria = {
+    field: SearchReleasesFields;
+    value: string;
+}[];
+
+class MusicBrainzUrlHelper {
+    public static getSearchReleasesQuery<T>(
+        searchFields: SearchReleasesCriteria,
+    ): string {
+        return `?query=${searchFields.map((f) => {
+            return `${f.field}:"${encodeURIComponent(f.value)}"`;
+        })}`;
+    }
+
+    public static getGeneralQueryString(
+        offset?: number,
+        limit?: number,
+    ): string {
+        let result = '';
+        if (offset !== undefined) {
+            result += `&offset=${offset}`;
+        }
+        if (limit !== undefined) {
+            result += `&limit=${limit}`;
+        }
+        result += '&fmt=json';
+        return result;
+    }
+}
+
+class MusicBrainzClient {
+    private apiEndpoint = 'https://musicbrainz.org/ws/2/';
+    /**
+     * @param searchFor query parameters to search for.
+     * @param limit how many entries should be returned. Only values between 1 and 100 (both inclusive) are allowed. If not given, this defaults to 25.
+     * @param offset search results starting at a given offset. Used for paging through more than one page of results.
+     */
+    public async searchReleases(
+        searchFor: SearchReleasesCriteria,
+        limit?: number,
+        offset?: number,
+    ) {
+        // https://musicbrainz.org/ws/2/recording?query=You%20Need%20Me%20AND%20artist=Ed%20sheeran&fmt=json&offset=0&limit=5
+
+        const endpoint = `${
```

Comment:
It's difficult to understand this.
Why not just extract the `${}` calls to separate variables?

`const searchQuery = MusicBranzUrlHelper.getSearch...`
`const generalQuery = MusicBrainzUrlHelper.getGeneral...`
and use those in the `const endpoint = ...`